dev-portal/README.md

```diff
@@ -1,24 +1,34 @@
 ## Development guide
 
-```sh
-# Generate a new build from your assets...
-npm run build
+### npm Scripts
 
-# Package all assets via the SAM cli
-sam package --template-file ./cloudformation/template.yaml --output-template-file packaged.yaml --s3-bucket my-cool-dev-portal-build-assets
+### `build` -- `npm run build`
 
-# Deploy the packaged assets.
-sam deploy --template-file ./packaged.yaml --stack-name "my-cool-dev-portal" --capabilities CAPABILITY_NAMED_IAM --parameter-overrides DevPortalSiteS3BucketName="my-cool-dev-portal-site-assets" ArtifactsS3BucketName="my-cool-dev-portal-api-assets"
+Builds a production-quality build and copies it to the lambda uploader.
 
-# get the outputs
-aws cloudformation describe-stacks --stack-name my-cool-dev-portal
+### `deploy` -- `npm run deploy`
 
-# fill config.js with the appropriate values listed in the 'outputs'
-```
+Starts a `sam package` and `sam deploy`. Does not build, but does write the config after a successful deploy.
+
+### `write-config` -- `npm run write-config`
+
+Gets the output from the CFN stack and writes a local version of the config.js file, so that you can run locally.
+
+### `release` -- `npm run release`
+
+Synonymous with `npm run build; npm run deploy; npm run write-config`.
 
-To use the build script, create a `build.config.js` file inside dev-portal.
+### `start` -- `npm run start`
+
+Starts up the local development server. Must have a `config.js` file locally for this to work.
+
+### If you have already deployed a Development Portal...
+
+1. Create a `build.config.js` file inside `/dev-portal/` with the following structure
```

Comment:
This could also probably be done by making a describe-stacks call and using the information there (from outputs, mostly).

Reply:
Is it possible to "pipe" the input params back to outputs? 'Cause that would solve that problem. We could just ask for stack-name, store it and the describe to build the rest of the file (or we could do that call every time). 

Thinking beyond that, we could auto-generate buckets for them too. (e.g. `${stackName}-${FirstSecondOrThirdBucket}-${guid}`).

The sad bit is that this is only usable by people who do the release/deployment manually from the dev-portal project.

Reply:
It very much is possible to just ref an input back to an output.